internal/lib/container_server.go

```diff
@@ -595,22 +603,26 @@ func (c *ContainerServer) HasContainer(id string) bool {
 }
 
 // RemoveContainer removes a container from the container state store
-func (c *ContainerServer) RemoveContainer(ctr *oci.Container) {
+func (c *ContainerServer) RemoveContainer(ctx context.Context, ctr *oci.Container) {
+	ctx, span := log.StartSpan(ctx)
+	defer span.End()
 	sbID := ctr.Sandbox()
 	sb := c.state.sandboxes.Get(sbID)
 	if sb == nil {
 		return
 	}
-	sb.RemoveContainer(ctr)
+	sb.RemoveContainer(ctx, ctr)
 	c.RemoveStatsForContainer(ctr)
 	if err := ctr.RemoveManagedPIDNamespace(); err != nil {
-		logrus.Errorf("Failed to remove container %s PID namespace: %v", ctr.ID(), err)
+		logrus.WithContext(ctx).Errorf("Failed to remove container %s PID namespace: %v", ctr.ID(), err)
```

Comment:
same here (and all cases): instead of using WithContext, I think we should just use our internal log package